File: MangoActuator/src/network/http/system/tasks_details.py
# -*- coding: utf-8 -*-
# @Project: 芒果测试平台
# @Description: 
# @Time   : 2024-09-24 10:01
# @Author : 毛鹏
from src.network.http.http_base import HttpBase


class TasksDetails(HttpBase):
    _url = 'system/tasks/details'

    @classmethod
    def get_tasks_list(cls, page, page_size, params: dict = None):
        _params = {
            'page': page,
            'pageSize': page_size
        }
        if params:
            _params.update(params)
        return cls.get(TasksDetails._url, params=_params)

    @classmethod
    def post_tasks_list(cls, json_data: dict):
        return cls.post(TasksDetails._url, json=json_data)

    @classmethod
    def put_tasks_list(cls, json_data: dict):
        return cls.put(TasksDetails._url, json=json_data)

    @classmethod
    def delete_tasks_list(cls, _id, ):
        return cls.delete(TasksDetails._url, params={'id': _id, })

    @classmethod
    def get_type_case_name(cls, _type, module_id):
        return cls.get(f'{TasksDetails._url}/type/case/name', params={
            'type': _type,
            'module_id': module_id
        })

    @classmethod
    def batch_set_cases(cls, case_id_list, tasks_id):
        return cls.post(f'{TasksDetails._url}/batch/set/cases', json={
            'case_id_list': case_id_list,
            'scheduled_tasks_id': tasks_id
        })

    @classmethod
    def put_tasks_case_test_object(cls, case_list, test_obj_id):
        return cls.put(f'{TasksDetails._url}/case/test/object', json={
            'case_list': case_list,
            'test_obj_id': test_obj_id
        })

File: mango-console/src/api/system/tasks_details.ts
import { deleted, get, post, put } from '@/api/http'

export function getSystemTasksRunCase(data: object) {
  return get({
    url: 'system/tasks/details',
    data: () => {
      return data
    },
  })
}

export function postSystemTasksRunCase(data: object) {
  return post({
    url: 'system/tasks/details',
    data: () => {
      return data
    },
  })
}
export function putSystemTasksRunCase(data: object) {
  return put({
    url: 'system/tasks/details',
    data: () => {
      return data
    },
  })
}

export function deleteSystemTasksRunCase(id: number | string[] | number[]) {
  return deleted({
    url: 'system/tasks/details',
    data: () => {
      return {
        id: id,
      }
    },
  })
}
export function postSystemTasksBatchSetCases(
  caseIdList: string[],
  scheduledTasksId: any,
  type: number
) {
  return post({
    url: 'system/tasks/details/batch/set/cases',
    data: () => {
      return {
        case_id_list: caseIdList,
        scheduled_tasks_id: scheduledTasksId,
        type: type,
      }
    },
  })
}

export function getSystemTasksTypeCaseName(type: any, moduleId: number) {
  return get({
    url: 'system/tasks/details/type/case/name',
    data: () => {
      return {
        type: type,
        module_id: moduleId,
      }
    },
  })
}
